src/Validation/Validator.php

```diff
@@ -1320,6 +1320,56 @@ public function multipleOptions($field, array $options = [], $message = null, $w
     }
 
     /**
+     * Add a validation rule to ensure that a field is an array containing at least
+     * the specified amount of elements
+     *
+     * @param string $field The field you want to apply the rule to.
+     * @param int $count The number of elements the array should at least have
+     * @param string|null $message The error message when the rule fails.
+     * @param string|callable|null $when Either 'create' or 'update' or a callable that returns
+     *   true when the validation rule should be applied.
+     * @see \Cake\Validation\Validation::numElements()
+     * @return $this
+     */
+    public function hasAtLeast($field, $count, $message = null, $when = null)
```

Comment:
The code of ``hasAtLeast()`` and ``hasAtMost()`` almost the same.
Besides the name of the method, only two things differ:
1. ``'hasAtLeast'`` vs. ``'hasAtMost'``
2. ``'>=' ``vs. ``'<='``

I wonder if that could be de-duplicated somehow.